src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,44 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<? extends Machine> getAll(Class type) {
+        List<? extends Machine> machines = new ArrayList<>();
+        if (Bulldozer.class == type) {
+            BulldozerProducer bulldozerProducer = new BulldozerProducer();
+            machines = bulldozerProducer.get();
```

Comment:
```suggestion
            return new ArrayList<>(bulldozerProducer.get());
```
Fix it in other cases